controllers/nstemplateset/nstemplateset_controller.go

```diff
@@ -232,7 +238,75 @@ func listByOwnerLabel(username string) runtimeclient.ListOption {
 	return runtimeclient.MatchingLabels(labels)
 }
 
-func isUpToDateAndProvisioned(obj metav1.Object, tierTemplate *tierTemplate) bool {
-	return obj.GetLabels()[toolchainv1alpha1.TemplateRefLabelKey] != "" &&
+// isUpToDateAndProvisioned checks if the obj has the correct Template Reference Label.
+// If so, it processes the tier template to get the expected roles and rolebindings and then checks if they are actually present in the namespace.
+func isUpToDateAndProvisioned(obj metav1.Object, tierTemplate *tierTemplate, r *namespacesManager) (bool, error) {
+	isLabelCorrect := obj.GetLabels()[toolchainv1alpha1.TemplateRefLabelKey] != "" &&
```

Comment:
no need to evaluate and store in a var (but that's a minor thing)

Reply:
done in 224a17d